tests/test_views.py

```diff
@@ -0,0 +1,927 @@
+            {'search-url-value': url.entity, 'partial-search': ''}
+        )
+
+        assert response.context['project'] == project
+        assert len(response.context['url_list']) == 1
+        assert response.context['url_list'][0] == url
+
+
+class TestProjectUrls():
+
+    def test_status_ok(self, client):
+        project = factories.ProjectFactory()
+        response = client.get(reverse('project_urls', args=[project.project_slug]))
+        assert response.status_code == 200
+
+    def test_template_used(self, client):
+        project = factories.ProjectFactory()
+        response = client.get(reverse('project_urls', args=[project.project_slug]))
+        assert response.templates[0].name == 'nomination/project_urls.html'
+
+    def test_context(self, client):
+        project = factories.ProjectFactory()
+        factories.URLFactory.create_batch(
+            3,
+            url_project=project,
+            attribute='surt',
+            value='http://(com,example.www)'
+        )
+        factories.NominatedURLFactory.create_batch(
+            2,
+            url_project=project,
+            value=1
+        )
+        response = client.get(reverse('project_urls', args=[project.project_slug]))
+
+        assert response.context['project'] == project
+        assert response.context['url_count'] == 3
+        assert response.context['nominator_count'] == 2
+        assert len(response.context['browse_tup']) == 1
+
+
+class TestUrlListing():
+
+    @pytest.mark.parametrize('request_type', [
+        'get',
+        'post'
+    ])
+    def test_status_ok(self, client, request_type):
+        entity = 'www.example.com'
+        project = factories.ProjectFactory()
+        factories.URLFactory(url_project=project, entity=entity)
+        response = getattr(client, request_type)(
+            reverse('url_listing', args=[project.project_slug, entity]))
+
+        assert response.status_code == 200
+
+    @pytest.mark.parametrize('request_type', [
+        'get',
+        'post'
+    ])
+    def test_status_ok_when_url_does_not_exist(self, client, request_type):
+        entity = 'www.example.com'
+        project = factories.ProjectFactory()
+        response = getattr(client, request_type)(
+            reverse('url_listing', args=[project.project_slug, entity]))
+
+        assert response.status_code == 200
+
+    @pytest.mark.parametrize('request_type, expected_template', [
+        ('get', 'nomination/url_listing.html'),
+        ('post', 'nomination/url_listing.html')
+    ])
+    def test_template_used(self, client, request_type, expected_template):
+        entity = 'www.example.com'
+        project = factories.ProjectFactory()
+        factories.URLFactory(url_project=project, entity=entity)
+        response = getattr(client, request_type)(
+            reverse('url_listing', args=[project.project_slug, entity]))
+
+        assert response.templates[0].name == expected_template
```

Comment:
ah, nope, that doesn't need to be a parameter anymore. same for the test below. I'll remove those from the parametrizations.